### DISCUSSION TITLE
How does Isar handle added properties in schema migration?

### DISCUSSION THREAD

**Discussion by davorkomusanac** [isar/isar#550] [Help]
Hi, I am just wondering, if I already have built a schema, and filled it with my data, and now I wish to add additional properties to my data class. How exactly does Isar handle that? 

I am asking because I added the additional properties, and put them as e.g. "late String description", and I see that inside my app it's value is an empty string, while Isar Inspector shows me the value is actually Null. 

Does that mean that Isar automatically converts all null values to some empty value? I am just curious, thanks

**Comment by simc** [comment]
Yes Isar will assign a default value to new non-nullable properties.

The next version will allow you to customize the default value through the constructor
